src/Bicep.LangServer/Server.cs

```diff
@@ -86,6 +95,7 @@ private static void RegisterServices(CreationOptions creationOptions, IServiceCo
             services.AddSingleton<IResourceTypeProvider>(services => creationOptions.ResourceTypeProvider ?? AzResourceTypeProvider.CreateWithAzTypes());
             services.AddSingleton<ISnippetsProvider>(services => creationOptions.SnippetsProvider ?? new SnippetsProvider());
             services.AddSingleton<IFileResolver>(services => creationOptions.FileResolver ?? new FileResolver());
+            services.AddSingleton<ITelemetryProvider>(services => creationOptions.TelemetryProvider ?? new TelemetryProvider());
```

Comment:
Is there any reason to provide this flexibility? Feels like it would be more straightforward to remove TelemetryProvider from creationOptions, and just set it up with:
```csharp
services.AddSingleton<ITelemetryProvider, BicepTelemetryProvider>();
```

Reply:
Updated here:
https://github.com/Azure/bicep/commit/ffbef4cf981f55f6966f50de6ed4dec09bd90905